examples/MultiAxisExample.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Add path to library (just for examples; you do not need this)
+import initExample
+import numpy as np
+import pyqtgraph as pg
+from pyqtgraph.Qt import QtGui
+
+app = pg.mkQApp()
+mw = QtGui.QMainWindow()
+mw.resize(800, 800)
+pg.setConfigOption("background", "w")
+pg.setConfigOption("foreground", "k")
+
+mpw = pg.MultiAxisPlotWidget()
+
+mw.setCentralWidget(mpw)
+mw.show()
+
+# LEGEND
+mpw.addLegend(offset=(0, 0))
+# TITLE
+mpw.setTitle("MultiAxisPlotWidget Example")
+# AXYS
+ax1 = mpw.addAxis("sx1", "bottom", text="Samples1", units="sx1")
+ax2 = mpw.addAxis("sx2", "bottom", text="Samples2", units="sx2")
+ax3 = mpw.addAxis("sx3", "bottom", text="Samples3", units="sx3")
+ay1 = mpw.addAxis("sy1", "left", text="Data1", units="sy1")
+ay2 = mpw.addAxis("sy2", "left", text="Data2", units="sy2")
+ay3 = mpw.addAxis("sy3", "left", text="Data3", units="sy3")
+# CHARTS
+c1, pi1 = mpw.addChart("Dataset 1", x_axis="sx1", y_axis="sy1")
+c2, pi2 = mpw.addChart("Dataset 2", x_axis="sx2", y_axis="sy1")
+c3, pi3 = mpw.addChart("Dataset 3", x_axis="sx2", y_axis="sy2")
+c4, pi4 = mpw.addChart("Dataset 4", x_axis="sx3", y_axis="sy3")
+# make and display chart
+mpw.makeLayout()
```

Comment:
I'm wondering whether we should do away with the need to call `makeLayout`; it seems unnecessary. For prior works, `addItem` has immediate effect, so we could reasonably say charts and axes get added to the visible layout immediately, and I don't think anyone would be surprised. We can leave `makeLayout` as a useful way to change that layout afterwards.